### DISCUSSION TITLE
Add ability to show how pylance came up with the type it thinks something is.

### DISCUSSION THREAD

**Discussion by rchiodo** [microsoft/pylance-release#3953] [Enhancement]
Was talking with a 'customer' last night. He said he hated Pylance because it never had completions that worked. 

I explained this might be because Pylance thought the type of whatever he was using was Any or Unknown. 

His response was `How does it know that? Where is it deciding the type is Any? If it could show me the path it took to compute the type, I could fix it`.

This feature request is to have some way to show that path through the code that Pylance (Pyright) used to determine the type of something.

**Comment by bschnurr** [comment]
Would be nice to provide a source file link clickable to go to function definition.

or a peek like functionality to view it.
